### ISSUE TITLE
[Bug] - Error while SELECT on VIEW with an ORDER BY referencing a renamed OBJECT

### ISSUE THREAD

**Issue by proddata** [crate/crate#11049]
**CrateDB version**:
CrateDB 4.4.1

**Environment description**:
CrateDB Cloud Cluster (3 node)

**Problem description**:

When creating a view with an `ORDER BY` using a renamed reference to an object, the SELECT on that view fails with an Error.

**Steps to reproduce**:

```sql
CREATE TABLE IF NOT EXISTS "r"."t"
(
    "ts"                    TIMESTAMP WITH TIME ZONE,
    "g_ts_month"            TIMESTAMP WITH TIME ZONE GENERATED ALWAYS AS date_trunc('month', "ts"),
    "obj_strict"            OBJECT(STRICT) AS (
        "timeStamp" TIMESTAMP WITH TIME ZONE
        )
) PARTITIONED BY ("g_ts_month")
```

```sql
CREATE OR REPLACE VIEW v_test AS
SELECT "obj_strict"                   "p_v"
FROM "r"."t"
ORDER BY "obj_strict"['timeStamp'] DESC;
```

```sql
SELECT *
FROM v_test
LIMIT 3;
```

```sql
ERROR: Can't handle Symbol [Reference: obj_strict]
```

---
Removing the rename works:
```sql
CREATE OR REPLACE VIEW v_test AS
SELECT "obj_strict"          
FROM "r"."t"
ORDER BY "obj_strict"['timeStamp'] DESC;
```
Also removing the order by in the view works:
```sql
CREATE OR REPLACE VIEW v_test AS
SELECT "obj_strict"                   "p_v"
FROM "r"."t";
```

adding the field used in order by also works:
edit:
```sql
CREATE OR REPLACE VIEW v_test AS
SELECT "obj_strict"                   "p_v",
"obj_strict"['timeStamp']
FROM "r"."t"
ORDER BY "obj_strict"['timeStamp'] DESC;
```

**Comment by proddata** [comment]
Seems to be something introduced with 4.2 or later

**Comment by seut** [comment]
@proddata Thanks fir reporting this, we can reproduce it and working on a fix.

Btw. the `limit 3` is the problematic part here as this results in a query-then-fetch execution plan. There is an issue with applying the fetch rewrite correctly when a column is aliased.

**Comment by seut** [comment]
@proddata We've fixed this by #11067, it will be part of the upcoming 4.4.2 release.

**Comment by proddata** [comment]
awesome 😄
